pkg/cmd/build/logs.go

```diff
@@ -57,12 +61,52 @@ var buildLogsCmd = &cobra.Command{
 		}
 
 		stopLogs := false
-		apiclient_util.ReadBuildLogs(activeProfile, buildId, query, &stopLogs)
+		go apiclient_util.ReadBuildLogs(activeProfile, buildId, query, &stopLogs)
+
+		if !continueOnCompletedFlag {
+			err = waitForBuildToComplete(buildId, apiClient)
+			if err != nil {
+				log.Fatal(err)
+			}
+			stopLogs = true
+		} else {
+			// Sleep indefinitely
+			select {}
+		}
+
+		// Make sure the terminal cursor is reset
+		fmt.Print("\033[?25h")
 	},
 }
 
+func waitForBuildToComplete(buildId string, apiClient *apiclient.APIClient) error {
+	for {
+		build, res, err := apiClient.BuildAPI.GetBuild(context.Background(), buildId).Execute()
+		if err != nil {
+			return apiclient_util.HandleErrorResponse(res, err)
+		}
+
+		completedStates := []apiclient.BuildBuildState{
+			apiclient.BuildStatePublished,
+			apiclient.BuildStateError,
+			apiclient.BuildStatePendingDelete,
```

Comment:
`PendingDelete` shouldn't be a completion state. The runner might write some logs after this point so we should def continue.